ocaml/typing/uniqueness_analysis.ml

```diff
@@ -1020,26 +1040,39 @@ let rec pattern_match_tuple pat values =
 and pattern_match_single pat paths : Ienv.Extension.t * UF.t =
   let loc = pat.pat_loc in
   let occ = Occurrence.mk loc in
+  (* To read from the allocation, we need to borrow its memory cell
```

Comment:
In the `` `NoRead `` case, the unique barrier will never be used again, right? Then we can immediately set it to `Resolved`, I think. (It's not that this is more efficient -- it's that it makes the code easier to understand.)